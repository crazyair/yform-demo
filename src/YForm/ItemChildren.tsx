import React from 'react';
import { Form } from 'antd';
import { forEach, omit } from 'lodash';
import { FormItemProps } from 'antd/lib/form';
import { YFormItemProps } from './Items';

export default (props: YFormItemProps) => {
	const { children, addonAfter, ...rest } = props;
	let _required: boolean | undefined = false;
	// 根据 rules required 判断外部 Form.Item 是否必填
	forEach(props.rules, item => {
		if ('required' in item) _required = item.required;
	});

	return children ? (
		<Form.Item required={_required} {...omit(rest, ['name', 'rules', 'dependencies', 'shouldUpdate'])}>
			<Form.Item noStyle {...rest}>
				{children as FormItemProps['children']}
			</Form.Item>
			<>{addonAfter}</>
		</Form.Item>
	) : null;
};
