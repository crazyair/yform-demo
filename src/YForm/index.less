.one-line {
    display: inline-block;
    width: 100%;
    .ant-form-item-label {
        float: left;
        // TODO 适用于 input 高度 32px
        padding-top: 5px;
    }
    transition: all 0.3s;
}
.list {
    .padding-icons {
        span + span {
            padding-left: 8px;
        }
        font-size: 14px;
        padding-left: 8px;
    }
    .inline-icons {
        position: absolute;
        // TODO 适用于 input 高度 32px
        top: 5px;
    }
}

.dib {
    display: inline-block;
}

.can-input-length {
    position: relative;
    margin-bottom: 14px;
    .length {
        color: #999999;
        font-size: 12px;
        line-height: 1;
        position: absolute;
        left: 3px;
        bottom: 0;
        transform: translate(0, 100%);
    }
}

.input-money {
    .zh {
        font-size: 14px;
        line-height: 20px;
    }
}

.form-spin {
    text-align: center;
    padding: 30px 50px;
    margin: 20px 0;
}

.mb0 {
    margin-bottom: 0;
}
