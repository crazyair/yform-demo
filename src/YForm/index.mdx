---
name: YForm
route: /yform
order: 1
---

import JackBox2 from 'jackbox';
import Alert from './demo/demo1.tsx';
import { JackBox } from '../';

import Demo1CodeTsx from '!raw-loader!./demo/demo1.tsx';

## 代码演示

<JackBox tsCode={Demo1CodeTsx} demoName="默认用法" description="组件加载时立即执行">
    <Alert />
</JackBox>
