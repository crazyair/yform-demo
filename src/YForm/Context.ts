import React from 'react';
import { YFormProps } from './Form';
import { YFormItemsProps } from './Items';

export const YFormContext = React.createContext<YFormProps>({});

export const YFormItemsContext = React.createContext<YFormItemsProps>({});
