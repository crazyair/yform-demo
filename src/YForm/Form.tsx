import { Form, Spin } from 'antd';
import React, { useRef } from 'react';
import { merge } from 'lodash';

import { FormProps } from 'antd/lib/form';
import baseItemsType, { YFormItemsType } from './ItemsType';
import Items, { FormatFieldsValue, YFormItemProps } from './Items';
import { YFormContext } from './Context';

import { onFormatFieldsValue, submitFormatValues } from './utils';

export type YFormPluginsType = {
  placeholder?: boolean;
  required?: boolean;
  disabled?: boolean;
  labelLayout?: boolean;
  noLabelLayout?: boolean;
};

export type KeyValue = { [key: string]: any };
export type FieldsType<T> = { [K in keyof T]: string };

export interface YFormConfig {
  itemsType?: YFormItemsType;
}
let globalConfig: YFormConfig = {};

export const Config = (options: YFormConfig) => {
  globalConfig = merge({}, globalConfig, options);
};

export function useFormatFieldsValue<T = any>() {
  const formatFieldsValue = useRef([]);
  return {
    onFormatFieldsValue: onFormatFieldsValue<T>(formatFieldsValue.current),
    formatFieldsValue: formatFieldsValue.current,
  };
}

export interface YFormProps extends Omit<FormProps, 'children'> {
  isShow?: boolean;
  loading?: boolean;
  disabled?: boolean;
  required?: boolean;
  plugins?: YFormPluginsType | boolean;
  itemsType?: YFormItemsType;
  formatFieldsValue?: FormatFieldsValue[];
  children?: YFormItemProps['children'];
}

const InternalForm = (props: YFormProps) => {
  const {
    disabled,
    required,
    loading,
    itemsType,
    children,
    onFinish,
    formatFieldsValue,
    ...rest
  } = props;

  const _itemsTypeAll = {
    ...baseItemsType,
    ...itemsType,
    ...globalConfig.itemsType,
  } as YFormItemsType;
  const _props = { plugins: true, ...props, itemsType: _itemsTypeAll };
  if ('isShow' in props && !props.isShow) {
    return null;
  }
  if (loading) {
    return (
      <div className="form-spin">
        <Spin />
      </div>
    );
  }
  const handleOnFinish = (value: KeyValue) => {
    if (onFinish) {
      if (formatFieldsValue) {
        onFinish(submitFormatValues(value, formatFieldsValue));
      } else {
        onFinish(value);
      }
    }
  };

  return (
    <Form {...rest} onFinish={handleOnFinish}>
      <YFormContext.Provider value={_props}>
        <Items>{children}</Items>
      </YFormContext.Provider>
    </Form>
  );
};

export default InternalForm;
