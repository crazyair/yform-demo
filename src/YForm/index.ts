import { Form as AntForm } from 'antd';
import YForm, { Config, useFormatFieldsValue } from './Form';
import Items from './Items';
import ItemsType from './ItemsType';
import { submitFormatValues, onFormatFieldsValue } from './utils';

import './index.less';

type InternalYForm = typeof YForm;
interface RefYForm extends InternalYForm {
	config: typeof Config;
	ItemsType: typeof ItemsType;
	Items: typeof Items;
	Item: typeof AntForm.Item;
	useFormatFieldsValue: typeof useFormatFieldsValue;
	submitFormatValues: typeof submitFormatValues;
	onFormatFieldsValue: typeof onFormatFieldsValue;
}

const Form: RefYForm = YForm as RefYForm;

Form.ItemsType = ItemsType;
Form.config = Config;
Form.Items = Items;
Form.Item = AntForm.Item;
Form.useFormatFieldsValue = useFormatFieldsValue;
Form.submitFormatValues = submitFormatValues;
Form.onFormatFieldsValue = onFormatFieldsValue;

export default Form;
