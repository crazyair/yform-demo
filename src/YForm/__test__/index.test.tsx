// tslint:disable:no-console
import React from 'react';
import { render } from '@testing-library/react';
import { mount } from 'enzyme';

import YForm from '..';
import { layout, layoutMore } from '../../example/utils';
import { YFormProps } from '../Form';

const YFormDemo = (props: YFormProps) => {
    const onFinish = (values: any) => {
        console.log('Success:', values);
    };

    const { onFormatFieldsValue, formatFieldsValue } = YForm.useFormatFieldsValue();

    onFormatFieldsValue([{ name: ['demo', 'aa'], format: (value: { users: any }) => value.users }]);

    return (
        <YForm onFinish={onFinish} formatFieldsValue={formatFieldsValue} initialValues={{ money: '1' }} {...props}>
            {[
                { type: 'input', name: 'demo' },
                { type: 'money', name: 'money', label: 'money' },
                {
                    dataSource: [
                        {
                            type: 'button',
                            noStyle: true,
                            plugins: { disabled: false },
                            componentProps: { type: 'primary', htmlType: 'submit', children: 'submit' },
                        },
                    ],
                },
            ]}
        </YForm>
    );
};

describe('YForm', () => {
    test('renders', () => {
        const wrapper = render(<YFormDemo />);
        expect(wrapper).toMatchSnapshot();
    });
    test('isShow', () => {
        const wrapper = render(<YFormDemo isShow={false} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('loading', () => {
        const wrapper = render(<YFormDemo loading />);
        expect(wrapper).toMatchSnapshot();
    });
    test('layout', () => {
        const wrapper = render(
            <div>
                <YFormDemo {...layout} />
                <YFormDemo {...layoutMore} />
            </div>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('layout', () => {
        const itemsType: any = {
            demo: { component: <p>123</p>, formatStr: '请输入${label}' },
            render: { render: () => <div>1</div> },
        };
        YForm.config({ itemsType });
        const wrapper = render(<YForm name="basic">{[{ type: 'demo' }, { type: 'render' }] as any}</YForm>);
        expect(wrapper).toMatchSnapshot();
    });
    test('onFinish', () => {
        const wrapper = mount(<YFormDemo />);
        wrapper.find('Button').simulate('submit');
        const wrapper2 = mount(<YFormDemo formatFieldsValue={undefined} />);
        wrapper2.find('Button').simulate('submit');
    });
});
