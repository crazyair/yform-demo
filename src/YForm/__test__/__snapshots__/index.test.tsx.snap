// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YForm isShow 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`YForm layout 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="yform-items"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <input
                    class="ant-input"
                    id="demo"
                    placeholder="请输入"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-col-8 ant-form-item-label"
              >
                <label
                  class=""
                  title="money"
                >
                  money
                </label>
              </div>
              <div
                class="ant-col ant-col-16 ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="input-money"
                  >
                    <span
                      class="ant-input-affix-wrapper"
                    >
                      <input
                        class="ant-input"
                        id="money"
                        placeholder="请输入money"
                        type="text"
                        value="1"
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        元
                      </span>
                    </span>
                    <div
                      class="zh"
                    >
                      壹
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="yform-items"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        submit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="yform-items"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-24 ant-col-sm-20 ant-col-sm-offset-4"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <input
                    class="ant-input"
                    id="demo"
                    placeholder="请输入"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-4"
              >
                <label
                  class=""
                  title="money"
                >
                  money
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-20"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="input-money"
                  >
                    <span
                      class="ant-input-affix-wrapper"
                    >
                      <input
                        class="ant-input"
                        id="money"
                        placeholder="请输入money"
                        type="text"
                        value="1"
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        元
                      </span>
                    </span>
                    <div
                      class="zh"
                    >
                      壹
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-24 ant-col-sm-20 ant-col-sm-offset-4"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="yform-items"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        submit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="yform-items"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <input
                  class="ant-input"
                  id="demo"
                  placeholder="请输入"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class=""
                title="money"
              >
                money
              </label>
            </div>
            <div
              class="ant-col ant-col-16 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="input-money"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <input
                      class="ant-input"
                      id="money"
                      placeholder="请输入money"
                      type="text"
                      value="1"
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      元
                    </span>
                  </span>
                  <div
                    class="zh"
                  >
                    壹
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="yform-items"
                >
                  <button
                    class="ant-btn ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      submit
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="yform-items"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-24 ant-col-sm-20 ant-col-sm-offset-4"
            >
              <div
                class="ant-form-item-control-input"
              >
                <input
                  class="ant-input"
                  id="demo"
                  placeholder="请输入"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-4"
            >
              <label
                class=""
                title="money"
              >
                money
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-20"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="input-money"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <input
                      class="ant-input"
                      id="money"
                      placeholder="请输入money"
                      type="text"
                      value="1"
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      元
                    </span>
                  </span>
                  <div
                    class="zh"
                  >
                    壹
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-24 ant-col-sm-20 ant-col-sm-offset-4"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="yform-items"
                >
                  <button
                    class="ant-btn ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      submit
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`YForm layout2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ant-form ant-form-horizontal"
        id="basic"
      >
        <div
          class="yform-items"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <p>
                  123
                </p>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div>
                  1
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ant-form ant-form-horizontal"
      id="basic"
    >
      <div
        class="yform-items"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <p>
                123
              </p>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div>
                1
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`YForm loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-spin"
      >
        <div
          class="ant-spin ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-spin"
    >
      <div
        class="ant-spin ant-spin-spinning"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`YForm renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ant-form ant-form-horizontal"
      >
        <div
          class="yform-items"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <input
                  class="ant-input"
                  id="demo"
                  placeholder="请输入"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label"
            >
              <label
                class=""
                title="money"
              >
                money
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="input-money"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <input
                      class="ant-input"
                      id="money"
                      placeholder="请输入money"
                      type="text"
                      value="1"
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      元
                    </span>
                  </span>
                  <div
                    class="zh"
                  >
                    壹
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="yform-items"
                >
                  <button
                    class="ant-btn ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      submit
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="yform-items"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <input
                class="ant-input"
                id="demo"
                placeholder="请输入"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              title="money"
            >
              money
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="input-money"
              >
                <span
                  class="ant-input-affix-wrapper"
                >
                  <input
                    class="ant-input"
                    id="money"
                    placeholder="请输入money"
                    type="text"
                    value="1"
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    元
                  </span>
                </span>
                <div
                  class="zh"
                >
                  壹
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="yform-items"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="submit"
                >
                  <span>
                    submit
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
